File: ScannerApp/code/Token.js
class Token {
    // public String type;
    // public String value;
    // public boolean hasContent;

    constructor(type) {
        this.type = type;
        this.hasContent = false;
    }

    static TokenConstructFromValue(type, value) {
        let outToken = new Token(type);
        outToken.value = value;
        outToken.hasContent = true;
        return outToken;
    }

    toString() {
        if (this.hasContent)
            return "<" + this.type + "," + this.value + ">";
        else
            return "<" + this.type + ">";
    }
}

File: ScannerJava/src/Token.java
public class Token {
    public String type;
    public String value;
    public boolean hasContent;

    public Token(String type) {
        this.type = type;
        hasContent = false;
    }

    public Token(String type, String value) {
        this.type = type;
        this.value = value;
        hasContent = true;
    }

    public String toString() {
        if (hasContent)
            return "<" + type + "," + value + ">";
        else
            return "<" + type + ">";
    }
}
